### PR TITLE
MGMT-8796: clear msg about cidr range

### PR DESCRIPTION
**Pull request by slaviered**
CIDR block size should be large enough to accommodate all the hosts' IPs. When a user chooses a smaller range than required, the error message is not clear enough.

This PR adds a recommendation to the error message to guide the users.

## List all the issues related to this PR

- [ ] New Feature 
- [x] Enhancement 
- [x] Bug fix
- [ ] Tests
- [ ] Documentation
- [ ] CI/CD 

## What environments does this code impact?

- [ ] Automation (CI, tools, etc)
- [x] Cloud
- [x] Operator Managed Deployments
- [ ] None

## How was this code tested?

- [ ] assisted-test-infra environment
- [ ] dev-scripts environment
- [ ] Reviewer's test appreciated
- [x] Waiting for CI to do a full test run
- [ ] Manual (Elaborate on how it was tested)
- [ ] No tests needed

## Assignees

/cc @
/cc @

## Checklist

- [ ] Title and description added to both, commit and PR.
- [ ] Relevant issues have been associated (see [CONTRIBUTING] guide)
- [ ] Reviewers have been listed
- [ ] This change does not require a documentation update (docstring, `docs`, README, etc)
- [ ] Does this change include unit-tests (note that code changes require unit-tests)

## Reviewers Checklist

- Are the title and description (in both PR and commit) meaningful and clear?
- Is there a bug required (and linked) for this change?
- Should this PR be backported?

[Kubernetes community documentation]: https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#commit-message-guidelines
[CONTRIBUTING]: https://github.com/openshift/assisted-service/blob/master/CONTRIBUTING.md

### CODE REVIEW

**Comment by omertuc** [comment]
`IpInFreeList` list being `false` means that the VIP that was chosen is already in use by a different machine. "Please increase the CIDR range." is the wrong advise, it should be "please choose a VIP that's not in use by any machine"

Unless I misunderstood something

**Comment by slaviered** [comment]
@omertuc good catch. Fixing

File: internal/network/machine_network_cidr.go [hunk 1]
```diff
@@ -86,13 +86,13 @@ func VerifyVip(hosts []*models.Host, machineNetworkCidr string, vip string, vipN
 		return nil
 	}
 	if machineNetworkCidr == "" {
-		return errors.Errorf("%s <%s> cannot be set if Machine Network CIDR is empty", vipName, vip)
+		return errors.Errorf("%s <%s> cannot be set if Machine Network CIDR is empty. Please make sure that Machine Network CIDR is configured correctly", vipName, vip)
 	}
 	if !ipInCidr(vip, machineNetworkCidr) {
-		return errors.Errorf("%s <%s> does not belong to machine-network-cidr <%s>", vipName, vip, machineNetworkCidr)
+		return errors.Errorf("%s <%s> does not belong to machine-network-cidr <%s>. Try to increase the block range.", vipName, vip, machineNetworkCidr)
```
Comment (omertuc) [inline internal/network/machine_network_cidr.go:92]: Also here, I think the advice should be "pick a VIP that's within the machine network CIDR". Making the CIDR bigger will not necessarily help, e.g. if the VIP is completely foreign to it.

Also machine network is often something that the user has no control over, they can choose a different VIP but sometimes the machine network is outside their control

Comment (filanov) [reply internal/network/machine_network_cidr.go:92]: +1

File: internal/network/machine_network_cidr.go [hunk 2]
```diff
@@ -86,13 +86,13 @@ func VerifyVip(hosts []*models.Host, machineNetworkCidr string, vip string, vipN
 		return nil
 	}
 	if machineNetworkCidr == "" {
-		return errors.Errorf("%s <%s> cannot be set if Machine Network CIDR is empty", vipName, vip)
+		return errors.Errorf("%s <%s> cannot be set if Machine Network CIDR is empty. Please make sure that Machine Network CIDR is configured correctly", vipName, vip)
```
Comment (filanov) [inline internal/network/machine_network_cidr.go:89]: Not sure that this message is helpful, i mean the second part.

**Review by filanov — COMMENTED** [review]
Added comment in the ticket, suggesting to leave this validation as is and add another validation for the subnet size

File: internal/network/machine_network_cidr.go (hunk 1 above)
Comment (slaviered) [reply internal/network/machine_network_cidr.go:92]: scrapping this solution in favor of a range validation

**Comment by slaviered** [comment]
launched a new PR

**Comment by omertuc** [comment]
https://github.com/openshift/assisted-service/pull/3732

**Pull request closed without merging**